snuba/query/data_source/simple.py

```diff
@@ -31,6 +31,7 @@ class Entity(SimpleDataSource):
 
     key: EntityKey
     schema: ColumnSet
+    alias: Optional[str] = None
```

Comment:
The parser doesn't create join nodes, it creates entities, so I needed a way to bubble the alias/sample data up to the join parser.

Reply:
Why don't you return a Tuple or any dedicated data structure there siince this is a temporary information needed during parsing? If we keep the alias here we are likely to start seeing processing code that some times uses the alias from the entity and some times the one from IndividualNode in the join as it will be duplicated.

Reply:
Isn't this also necessary in single entity queries? We discussed that it is possible to alias an entity in the match clause even for single entity queries. Won't that information be needed in some way?

Reply:
> Isn't this also necessary in single entity queries?

That was not my understanding when we discussed this. That's why I kept it onlly in IndividualNode in the join. Exactly to avoid adding it for queries that should never have it.
My understanding about the discussion we had on when/how to apply table alias names to columns was related with joins, where fully qualifying the columns is needed.

If there is a good reason to allow individual entities in single queries to be aliased, I am ok with that (though i cannot think about one now). In any case, if we allow aliases in simple queries, we need to fully qualify all columns right away otherwise the query processing logic will have to deal with a mix of fully qualified columns and not fully qualified columns. Or in alternative we would have to remove all table aliases after parsing in simple queries (but why would we support this then ?)
This includes also columns added by query processors and extensions, which will become more complex since they will have to look into the data source to find the right table alias.

Reply:
OK, I have changed the spec. Single entity queries no longer accept an alias. Splitting that logic also allowed me to have the parser return join structures directly, so I also removed alias from the QueryEntity since I know return an IndividualNode instead.